src/translators/sql_translator.py

```diff
@@ -992,70 +992,53 @@ def _format_response(self, resultset, col_names, table_name, last_n):
 
         return [entities[k] for k in sorted(entities.keys())]
 
-    def delete_entity(self, entity_id, entity_type=None, from_date=None,
+    def delete_entity(self, eid, etype=None, from_date=None,
                       to_date=None, fiware_service=None,
                       fiware_servicepath=None):
-        if not entity_id:
+        if not eid:
             raise NGSIUsageError("entity_id cannot be None nor empty")
 
-        if not entity_type:
-            entity_type = self._get_entity_type(entity_id, fiware_service)
+        if not etype:
+            # TODO: rahter make entity type mandatory and bail out if not
+            # present.
+            # In fact, _get_entity_type will scan **all** DB tables to
+            # figure out candidate entity types for the given entity ID!
```

Comment:
this is also current orion logic, so i would not do differently

Reply:
oh dear. then we'll have to come up with an efficient implementation if we can't make the param mandatory in the API...let's leave this for another day. I'm going to open an issue for that and get rid of the todo comment if that's okay?

Reply:
k

Reply:
see #369